CMakePresets.json

```diff
@@ -0,0 +1,384 @@
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "windows-msvc-debug"
+        },
+        {
+            "name": "default-build-windows-msvc-release",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "windows-msvc-release"
+        },
+        {
+            "name": "verbose-build-windows-msvc-release",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "windows-msvc-release"
+        },
+        {
+            "name": "default-build-windows-clang-debug",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "windows-clang-debug"
+        },
+        {
+            "name": "verbose-build-windows-clang-debug",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "windows-clang-debug"
+        },
+        {
+            "name": "default-build-windows-clang-release",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "windows-clang-release"
+        },
+        {
+            "name": "verbose-build-windows-clang-release",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "windows-clang-release"
+        },
+        {
+            "name": "default-build-linux-gcc-debug",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "linux-gcc-debug"
+        },
+        {
+            "name": "verbose-build-linux-gcc-debug",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "linux-gcc-debug"
+        },
+        {
+            "name": "default-build-linux-gcc-release",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "linux-gcc-release"
+        },
+        {
+            "name": "verbose-build-linux-gcc-release",
+            "displayName": "Verbose",
+            "description": "Passes -v to Ninja",
+            "nativeToolOptions": [
+                "-v"
+            ],
+            "configurePreset": "linux-gcc-release"
+        },
+        {
+            "name": "default-build-linux-clang-debug",
+            "displayName": "Default",
+            "description": "Vanilla build",
+            "configurePreset": "linux-clang-debug"
+        },
```

Comment:
For this build, on Linux/CLion, I am getting a repeat of issue #40: CMake thinks it's using clang, but conan thinks it's using the default c++ compiler, which is g++. The fix proposed in #122 resolves this issue.

Whatever went wrong here for me might be a consequence of the way CLion implements support for CMakePresets: As discussed above, CLion only supports `buildPresets`.

I am a little confused by what the [CLion docs](https://www.jetbrains.com/help/clion/cmake-presets.html) mean when they say CLion "only supports `buildPresets`. When using this preset, CMake knows that it's supposed to use `clang`, `clang++`, and `ninja`. These tools are not listed in `buildPresets`, but they are listed in the `configurePreset` item specified here. Empirically, it appears that CLion supports `configurePresets` as well as `buildPresets`. Perhaps the docs are wrong?

Reply:
Also, I noticed that clang for windows in not able to build docopt.. It should be a conan bug?? (https://github.com/microsoft/vscode-cmake-tools/issues/2090) I don't know. I hope #122 fix that too. For now on windows local machine I can use only msvc toolchain.